### ISSUE TITLE
Dissociate fading and opacity steps

### ISSUE THREAD

**Issue by ThibaultNocchi** [yshui/picom#359]
### picom version
picom vgit-4c22d (the latest commit at the time of writing this)

### problem
I'm migrating from compton and encountering a problem on specifying the fading and the opacity steps. With compton I could specify an `alpha-step` of 0.06 and a `fade-in/out-step` with the default values. With picom `alpha-step` isn't supported so I'm having to choose between a reasonable timing when switching workspace but with no (seen) opacity fading between windows, or the opposite with a really slow workspace switch effect but a good opacity fading between windows. All of these by playing with the `fade-in/out-step` setting of picom.

I guess there is something I am missing, so if anyone could point me in the direction of the solution I'd be greatly appreciative. Thanks, have a nice day!

**Comment by yshui** [comment]
@ThibaultNocchi Sorry, I am not very sure what exactly is the effect you want to achieve.

picom uses the smallest possible alpha-step, i.e. for 8-bit color it's 1/256. Is that not what you want?

**Comment by ThibaultNocchi** [comment]
It seems I misunderstood the setting, sorry!

I'm using the setting to have the active window opaque and the others a little bit transparent. The "problem" I'm having is that under compton, I felt that the opacity change when changing the active window in the same workspace was a little bit smoother than under picom, and I don't know to what it's due.

I have prepared two videos comparing this under compton and picom.
[comparison.zip](https://github.com/yshui/picom/files/4417504/comparison.zip)

Thanks for your help!

**Comment by yshui** [comment]
@ThibaultNocchi And smaller fading steps in picom makes the transition smooth as well? This might be a bug...

**Comment by yshui** [comment]
Can you upload a `--log-level=trace` log file, generated when you do this?

**Comment by ThibaultNocchi** [comment]
Yes, putting a fading step of 0.01 makes the opacity transition more similar to what I'm used to compton, but the workspace switch fade is way slower.

How can I get the log file? When doing `./picom --log-level=trace > log` I get all the log in the terminal.

**Comment by yshui** [comment]
`./picom --log-level=trace --log-file=file`

**Comment by ThibaultNocchi** [comment]
Thanks! I've seen the log file is quite long, so I tried limiting my actions. I've only launched the software, switched a few times the active window and then quit. Thanks!

[log.txt](https://github.com/yshui/picom/files/4419776/log.txt)
